### PR TITLE
Displayer winning character

### PR DESCRIPTION
**Pull request by maxailloud**

### CODE REVIEW

File: js/states/score.js [hunk 1]
```diff
@@ -87,7 +114,34 @@ define([
...
-            // this.manager.createEntityFromAssemblage(winner.character);
+            var winningPlayer = this.manager.createEntityFromAssemblage(winner.character);
+            this.manager.updateComponentDataForEntity('Position', winningPlayer, {
+                x: 450,
+                y: 300
+            });
+            var characterData = this.manager.getComponentDataForEntity('Character', winningPlayer);
+            var playerData = this.manager.getComponentDataForEntity('Player', winningPlayer);
+            var displayableData = this.manager.getComponentDataForEntity('Displayable', winningPlayer);
+            displayableData.sprite = characterData.sprite + 'a';
```
Comment (adngdb) [inline js/states/score.js]: That would be surprising for the winner to see a character that is not his. Especially if both players chose the same character. Could you not load the correct sprite depending on the winning player's number?

**Comment by adngdb** [comment]
r+ when the correct character sprite is displayed.

**Comment by maxailloud** [comment]
What does "r+" mean ?

File: js/states/score.js (hunk 1 above)
Comment (maxailloud) [inline js/states/score.js]: I don't understand it, I didn't used the winning player's number.
I load the good sprite, where is the catch?

**Comment by adngdb** [comment]
`r+` means that I give you a positive review, so this PR is ready to be merged in my opinion. We'll need to sort out this problem first though. Let me test it a bit and see what's wrong.

File: js/states/score.js (hunk 1 above)
Comment (adngdb) [inline js/states/score.js]: So I just verified, if you just use `+ 'a'` here, it always loads the "blue" version of the character sprite. If player 2 wins, we want to load the "orange" version. 

In short, you should use something like that: https://github.com/AdrianGaudebert/nth/blob/master/js/states/game.js#L261

Comment (maxailloud) [inline js/states/score.js]: Ha yes you're right. I wrote it thinking "I will update that later", but I forgot.
I'll do it tonight, or at lunchtime.

**Comment by maxailloud** [comment]
Ok it should be good now.

There was a bug on manipulations attribution.
And the psirte for the player take care of the nega manipulation or not, to display the good one.

File: js/states/game.js [hunk 2]
```diff
@@ -262,7 +262,7 @@ function (
                 });
 
                 // Add the manipulations that were chosen by the other player.
-                this.manager.addComponentsToEntity(newPlayer, this.newPlayerManipulations[1 - player.number]);
+                this.manager.addComponentsToEntity(newPlayer, this.newPlayerManipulations[player.number]);
```
Comment (adngdb) [inline js/states/game.js:265]: Huh??? Are we not supposed to assign manipulations to the opponent ? Unless you store the list of manipulations in the opponent's component already? 

The way I see things is such:
- A and B choose a manipulation, and have that choice stored in their respective components
- A gets assigned B's manip, and B gets A's.

Isn't that what you did? In which case this line here was not a bug.

File: js/states/score.js [hunk 3]
```diff
@@ -86,8 +113,44 @@ define([
...
+                x: 450,
+                y: 300
+            });
+            var characterData = this.manager.getComponentDataForEntity('Character', winningPlayer);
+            var displayableData = this.manager.getComponentDataForEntity('Displayable', winningPlayer);
+            var playerData = this.manager.getComponentDataForEntity('Player', winningPlayer);
+            playerData.number = winner.number;
+
+            var spriteLetters = ['a', 'b'];
+            var spriteLetter = spriteLetters[playerData.number];
```
Comment (adngdb) [inline js/states/score.js:127]: Even simpler: `var spriteLetter = ['a', 'b'][playerData.number];`

**Comment by adngdb** [comment]
I'm confused about the manipulation storage. In my mind we stored the chosen manipulation in the data of the player who chose those manips, and then we applied them to the opponent. Did you do that the other way around?

File: js/states/game.js (hunk 2 above)
Comment (maxailloud) [inline js/states/game.js:265]: In this place it's not the choice of the manipulation but the assignation of the manipulations to the player at the beginning of a round.
It's for adding the Component let's say Nega to the player 1, as the player 2 choose it for him.

**Comment by maxailloud** [comment]
Yep exactly.
When a manipulation is choose I store it for the opponent directly, and in the game I just have to add manipulation to him, no need to add to the opponent as it's already done.

**Comment by adngdb** [comment]
OK good, I am much less confused now! 

r+ =)

**Pull request merged**